src/core/output-layout.cpp

```diff
@@ -1292,6 +1294,16 @@ class output_layout_t::impl
             remove_output(output);
         });
 
+        on_request_state.set_callback([=] (void *data)
+        {
+            auto ev = static_cast<wlr_output_event_request_state*>(data);
+            wlr_output_commit_state(ev->output, ev->state);
+            send_wlr_configuration();
+            outputs[output]->output->render->damage_whole();
```

Comment:
`outputs[output]->output` could be NULL for disabled outputs. I doubt wlroots will send a signal for such outputs but it is not forbidden so I would add a check for that.

Also fwiw if we send a wlr configuration, then we also need to emit the output-configuration-changed signal too.

Reply:
I have used `lo` instead of `outputs[output]->output` and installed the output-configuration-changed signal emission, which actually fixed a bug where resizing a second WL output and then the first would leave the second partially rendered until resizing it again.